src/client/MenuWindow.cpp

```diff
@@ -67,44 +70,60 @@ namespace client
             sf::Event event;
             while (clientMenuWindow->pollEvent(event))
             {
-
-                switch (event.type)
+                if (menuEventHappened(event))
                 {
-                case sf::Event::MouseButtonPressed:
-
-                    std::cout << "Click \n";
+                    return;
+                }
+            }
+        }
+    }
 
-                    break;
+    /*!
+     * \brief Test events and do actions corresponding to the event
+     * @param event pointer to the event
+     */
+    bool MenuWindow::menuEventHappened(sf::Event& event){
```

Comment:
It would have been easier to do only one event handling for all windows (it seems to be a c/p from GameWindow::gameEventHappened)

Reply:
The gameEngine should not use the SFML library, this is why we created it and I don't agree on adding the handler on the Engine. Moreover adding the handler in the engine will cause a huge adding of if and else loop to detect if the event happend when the user is on the menu or the game